website/docs/BestPractices.md

```diff
@@ -214,3 +214,76 @@ If abstracted away it could look something like below where the logic is put in
 // 👍
 await form.submitData(new Person('bob@webdriver.io'))
 ```
+
+## File Downloads
+
+When automating file downloads in web testing, it's essential to handle them consistently across different browsers to ensure reliable test execution. Here, we provide best practices for file downloads and demonstrate how to configure download directories for **Google Chrome**, **Mozilla Firefox**, and **Microsoft Edge**.
+
+### Download Paths
+
+**Hardcoding** download paths in test scripts can lead to maintenance issues and portability problems. Utilize **relative paths** for download directories to ensure portability and compatibility across different environments.
+
+```javascript
+// 👎
+// Hardcoded download path
+const downloadPath = '/path/to/downloads';
+
+// 👍
+// Relative download path
+const downloadPath = path.join(__dirname, 'downloads');
+```
+
+### Wait Strategies
+
+Failing to implement proper wait strategies can lead to race conditions or unreliable tests, especially for download completion. Implement **explicit** wait strategies to wait for file downloads to complete, ensuring synchronization between test steps.
+
+```javascript
+// 👎
+// No explicit wait for download completion
+await browser.pause(5000);
+
+// 👍
+// Wait for file download completion
+await waitUntil(async ()=> await fs.existsSync(downloadPath), 5000);
+```
+
+### Configuring Download Directories for Each Browser
```

Comment:
Can we also include changing the download directory through the CDP command via `getPuppeteer`? We might even work in Chrome, Edge and Firefox.

Reply:
@christian-bromann  Don't you think this should be the better practice rather than CDP? and cdp itself varies between firefox and chrome. so code won't be that flexible. What do you suggest?

Reply:
> What do you suggest?

I think at least for Chrome we should document this capability and note that this won't work cross browser.

Reply:
@christian-bromann have a look.